### PR TITLE
ENH: get rid of the defaults file

### PR DESCRIPTION
**Pull request by schacht-certat**
The defaults file was used to define global settings. This commit drops
the defaults file as a separate files and instead uses the 'global' key
in the runtime conf to define settings that should be effective in every
bot.
To accomplish that there is now a utils.get_global_settings() function
that extracts the 'global' dict from the runtime configuration file.

### CODE REVIEW

**Review by wagner-certat — CHANGES_REQUESTED** [review]
`defaults.conf` is still mentioned in the docs.

The file itself is also still there, do you plan to remove that in a separate PR? And then we need an upgrade function to
1) migrate the values (only those which differ them the internal defaults) to the runtime config
2) remove the file

**Comment by wagner-certat** [comment]
In https://github.com/certtools/intelmq/blob/01e8d0184b45d5247723844dd43c7d4d8e7b42da/intelmq/bin/intelmqctl.py#L1522 the "global" (non)bot needs to be skipped

**Review by waldbauer-certat — APPROVED** [review]

File: intelmq/lib/test.py [hunk 1]
```diff
@@ -73,6 +71,10 @@ def mocked(conf_file):
     return mocked
 
 
+def mocked_get_global_settings():
+    return BOT_CONFIG
```
[the remark this answers is not in the document]
Comment (schacht-certat) [reply intelmq/lib/test.py]: An empty dict does not work but I've reduced the values as much as possible.

**Comment by schacht-certat** [comment]
> `defaults.conf` is still mentioned in the docs.

fixed.

> 
> The file itself is also still there, do you plan to remove that in a separate PR?

fixed.

> And then we need an upgrade function to
> 
>     1. migrate the values (only those which differ them the internal defaults) to the runtime config
> 
>     2. remove the file

I added an upgrade function that removes the file and migrates the values, but I'm not sure how to find out how the values differ from internal defaults- the internal defaults might be distributed over multiple files...

**Comment by schacht-certat** [comment]
> In
> 
> https://github.com/certtools/intelmq/blob/01e8d0184b45d5247723844dd43c7d4d8e7b42da/intelmq/bin/intelmqctl.py#L1522
> the "global" (non)bot needs to be skipped

fixed

**Review by wagner-certat — CHANGES_REQUESTED** [review]
I'm missing docs in changelog (short) and the NEWS file (explanation to users, what happens on upgrade, e.g. values are migrated to runtime to global, user should check if some values are obsolete etc.)

File: docs/user/configuration-management.rst [hunk 2]
```diff
@@ -61,13 +60,6 @@ Use the IntelMQ Manager mentioned above to generate the configuration files if u
 
 In the shipped examples 4 collectors and parsers, 6 common experts and one output are configured. The default collector and the parser handle data from malware domain list, the file output bot writes all data to ``/opt/intelmq/var/lib/bots/file-output/events.txt``/``/var/lib/intelmq/bots/file-output/events.txt``.
 
-*******************************
```
[the remark this answers is not in the document]
Comment (schacht-certat) [reply docs/user/configuration-management.rst:64]: Fixed

File: docs/user/configuration-management.rst [hunk 3]
```diff
@@ -298,7 +290,7 @@ The configuration and the differences are outlined here.
 Keep in mind that it is slower, but has better monitoring capabilities and is more stable.
 The AMQP support is considered beta, so small problems might occur. So far, only RabbitMQ as broker has been tested.
 
-You can change the broker for single bots (set the parameters in the runtime configuration per bot) or for the whole botnet (in defaults configuration).
+You can change the broker for single bots (set the parameters in the runtime configuration per bot) or for the whole botnet.
```
[the remark this answers is not in the document]
Comment (schacht-certat) [reply docs/user/configuration-management.rst]: Fixed

File: NEWS.md [hunk 4]
```diff
@@ -41,6 +41,9 @@ The old names can still be used in code, and they are automatically converted to
 
 ### Configuration
 
+The `defaults.conf` file was removed. Settings that should effect all the bots are not part of the runtime.conf file and are configured in the `global` section in that file.
+The `intelmqctl upgrade` command migrates the existing values from the `defaults.conf` file to the `runtime.conf` file under the `global` section and then deletes the `defaults.conf` file.
```
Comment (wagner-certat) [inline NEWS.md:45]: It's `intelmqctl upgrade-config` :)

**Review by wagner-certat — CHANGES_REQUESTED** [review]

**Review by wagner-certat — APPROVED** [review]

**Pull request merged**